imap_processing/lo/l1b/lo_l1b.py

```diff
@@ -407,6 +415,36 @@ def set_each_event_epoch(l1b_de: xr.Dataset) -> xr.Dataset:
     return l1b_de
 
 
+def set_avg_spin_durations_per_event(
+    l1a_de: xr.Dataset, l1b_de: xr.Dataset, avg_spin_durations_per_cycle: xr.DataArray
+) -> xr.DataArray:
+    """
+    Set the average spin duration for each direct event.
+
+    Parameters
+    ----------
+    l1a_de : xarray.Dataset
+        The L1A DE dataset.
+    l1b_de : xarray.Dataset
+        The L1B DE dataset.
+    avg_spin_durations_per_cycle : xarray.DataArray
+        The average spin duration for each spin epoch.
+
+    Returns
+    -------
+    l1b_de : xarray.Dataset
+        The L1B DE dataset with the average spin duration added.
+    """
+    # repeat the average spin durations for each cycle based on the direct event count
+    # to get an average spin duration for each direct event. This will be used in L1C
+    # to calculate the exposure time for each direct event.
```

Comment:
This could be in the extended docstring too so it would be rendered in the documentation.

Is my understanding correct that `l1a_de["de_count"]` is number of de's per cycle? So this calculation is saying if I had 2 cycles of 3 and 4 events the output array would be: "cycle1, cycle1, cycle1, cycle2, cycle2, cycle2, cycle2"?

Reply:
That's correct. I'll add it to the docstring